Makefile

```diff
@@ -951,7 +951,111 @@ ifeq "$(OPENMP)" "true"
 endif
 
 
-pio_test:
+openacc_test : 
+ifeq "$(OPENACC)" "true"
+	@#
+	@# First ensure that both FFLAGS_ACC and CFLAGS_ACC are not blank
+	@# If these are not set for a target, then OpenACC most likely cannot compile
+	@#
+	@echo "Checking if FFLAGS_ACC and CFLAGS_ACC are defined for [$(BUILD_TARGET)]..."
+	@if ([ -z "$(FFLAGS_ACC)" ] && [ -z "$(CFLAGS_ACC)" ]); then \
+	   (echo "*********************************************************"; \
+	    echo "ERROR: OPENACC=true was specified, but [$(BUILD_TARGET)] build target does not seem to support OpenACC:"; \
+	    echo "    FFLAGS_ACC and CFLAGS_ACC are both undefined for [$(BUILD_TARGET)] in the top-level Makefile."; \
+	    echo "Please set these variables to appropriate OpenACC compilation flags in the [$(BUILD_TARGET)] target to enable OpenACC support."; \
+	    echo "*********************************************************"; exit 1)\
+	else \
+	   echo "=> FFLAGS_ACC or CFLAGS_ACC defined"; \
+	fi
+
+	@#
+	@# Create test C and FORTRAN programs that look for OpenACC header file and parallelize a loop
+	@#
+	@printf "#include <openacc.h>\n \
+	         &int main(){\n \
+	         &    int n_devs=acc_get_num_devices( acc_device_current );\n \
+	         &    int i,n=0;\n \
+	         &    #pragma acc kernels\n \
+	         &    for (i=0; i<10; i++)\n \
+	         &      n=n+i;\n \
+	         &    return 0;\n \
+	         &}\n" | sed 's/^[ \t]*&//' > openacc.c
+	@printf "program openacc\n \
+		&    use openacc\n \
+		&    integer :: i,n=0,n_devs=0\n \
+		&    n_devs=acc_get_num_devices( acc_device_current )\n \
+		&    !\$$acc kernels\n \
+		&    do i=0,10\n \
+		&      n=n+i\n \
+		&    end do\n \
+		&    !\$$acc end kernels\n \
+		&end program\n" | sed 's/^[ \t]*&//' > openacc.f90
+
+	@#
+	@# See whether the test programs can be compiled
+	@#
+	@echo "Checking [$(BUILD_TARGET)] compilers for OpenACC support..."
+	@$(SCC) openacc.c $(CPPINCLUDES) $(CFLAGS) $(LDFLAGS) $(LIBS) -o openacc_c.out > openacc_c.log 2>&1 \
+	  && echo "=> $(SCC) can compile test OpenACC program" || \
+	 (echo "*********************************************************"; \
+	  echo "ERROR: Test OpenACC C program could not be compiled by $(SCC)."; \
+	  echo "Following compilation command failed with errors:" ; \
+	  echo "$(SCC) openacc.c $(CPPINCLUDES) $(CFLAGS) $(LDFLAGS) $(LIBS) -o openacc_c.out"; \
+	  echo ""; \
+	  echo "Test program openacc.c and output openacc_c.log have been left"; \
+	  echo "in the top-level MPAS directory for further debugging"; \
+	  echo "*********************************************************"; \
+	  rm -rf openacc.f90 openacc_f.out openacc_f.log; exit 1)
```

Comment:
`openacc_f.out` and `openacc_f.log` should never exist at this point, right? Or is the idea to remove any failed Fortran test output from a previous build of MPAS?
Also, I think the `r` option may not be needed in the `rm` command.

Reply:
Yes to the second question. My idea with these `rm` commands after the error message was that only the relevant files will be left after a failed build. So on a first build, these files shouldn't exist but they may be there if someone had a failed build and adjusted their inputs to make. That said, I do think I need to change such `rm` commands a little so they remove any openacc_[cf].out that might exist. Those would also be possible extraneous files from the build.

I agree to the `r` option being unneeded. That will be removed.